dkg/pedersen/controller/mod.go

```diff
@@ -0,0 +1,55 @@
+package controller
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"go.dedis.ch/dela/cli"
+	"go.dedis.ch/dela/cli/node"
+	"go.dedis.ch/dela/dkg"
+	"go.dedis.ch/dela/dkg/pedersen"
+	"go.dedis.ch/dela/mino"
+	"golang.org/x/xerrors"
+)
+
+// NewMinimal returns a new minimal initializer
+func NewMinimal() node.Initializer {
+	return minimal{}
+}
+
+// minimal is an initializer with the minimum set of commands. Indeed it only
+// creates and injects a new DKG
+//
+// - implements node.Initializer
+type minimal struct{}
+
+// Build implements node.Initializer. In this case we don't need any command.
+func (m minimal) SetCommands(_ node.Builder) {
+}
+
+// Inject implements node.Initializer. It creates and registers a pedersen DKG.
+func (m minimal) Inject(ctx cli.Flags, inj node.Injector) error {
+	var no mino.Mino
+	err := inj.Resolve(&no)
+	if err != nil {
+		return xerrors.Errorf("failed to resolve mino: %v", err)
+	}
+
+	privkey := dkg.Suite.Scalar().Pick(dkg.Suite.RandomStream())
+	pubkey := dkg.Suite.Point().Mul(privkey, nil)
+
+	dkg := pedersen.NewPedersen(privkey, no)
```

Comment:
Instead of asking for the private key, you should let the constructor generates it (and later we can improve to load an existing one from a file or whatever). Then you can remove the public _Suite_ variable.

Reply:
mmm thing is the use need the associated public key. So are you suggesting the constructor also return the generated public key ?